Core/CoreTests/AppEnvironment/LocalizationManagerTests.swift

```diff
@@ -46,7 +46,31 @@ class LocalizationManagerTests: CoreTestCase {
         XCTAssertTrue(LocalizationManager.needsRestart)
     }
 
-    func testSetCurrentLocale() {
+    func testConvertCustomLocale() {
+        var inputLocale: String?
+
+        inputLocale = "pt-BR"
+        XCTAssertEqual(LocalizationManager.convertCustomLocale(inputLocale), "pt-BR")
+
+        inputLocale = "en-AU-x-unimelb"
+        XCTAssertEqual(LocalizationManager.convertCustomLocale(inputLocale), "en-AU-unimelb")
+
+        inputLocale = "da-x-k12"
+        XCTAssertEqual(LocalizationManager.convertCustomLocale(inputLocale), "da-instk12")
+
+        inputLocale = "en-AU-x-123456789012345"
+        XCTAssertEqual(LocalizationManager.convertCustomLocale(inputLocale), "en-AU-12345678")
+
+        inputLocale = nil
+        XCTAssertEqual(LocalizationManager.convertCustomLocale(inputLocale), "")
+    }
+
+    // This test is disabled, because Language & Region are fixed (en & US) for tests, and it would always fail.
+    // Main logic is extracted and tested above.
+    // To enable it: set TestPlan (or Scheme) Language & Region to System, comment out XCTSkip() below.
+    func testSetCurrentLocale() throws {
+        try XCTSkipIf(true, "This test is disabled, because Language & Region are fixed (en & US) for tests, and it would always fail.")
+
```

Comment:
Does it make sense to keep this test at all if it's disabled? Who's going to enable it?

Reply:
The idea was, that if the related product code will be changed in the future, it may be a good idea to test it.